arangod/Cache/Cache.cpp

```diff
@@ -343,7 +343,10 @@ void Cache::shutdown() {
       }
 
       SpinUnlocker taskUnguard(SpinUnlocker::Mode::Write, _taskLock);
-      std::this_thread::sleep_for(std::chrono::microseconds(10));
+      SpinUnlocker shutdownUnguard(SpinUnlocker::Mode::Write, _shutdownLock);
+
+      // sleep a bit without holding the locks
+      std::this_thread::sleep_for(std::chrono::microseconds(20));
```

Comment:
Why do we need acquire this spinlocks before first cycle iteration, but we don't need to require afrer cycle gone or on next operation?

In general looks like bug.

Reply:
Mmh, don't understand.
We have the spinlock acquired for each iteration in that loop.

Reply:
Can you name line number for me?
I don't see where lock acquired in cycle.

Reply:
Hmm or Unlocker call unlock in ctor and then call lock in dtor?

Ok I understand now.